Core.ApplicationServices/References/ReferenceService.cs

```diff
@@ -200,22 +200,43 @@ private void RaiseRootUpdated(IEntityWithExternalReferences owner)
                             case > 1:
                                 return new OperationError("Only one reference can be master reference", OperationFailure.BadInput);
                         }
-
-                        foreach (var referenceProperties in newReferences)
+                        
+                        var referencesToUpdate = new List<ExternalReference>();
+                        foreach (var externalReferenceProperties in referenceList)
                         {
-                            var result = AddReference(rootId, rootType, referenceProperties.Title, referenceProperties.DocumentId, referenceProperties.Url);
-
-                            if (result.Failed)
-                                return new OperationError($"Failed to add reference with data:{JsonConvert.SerializeObject(referenceProperties)}. Error:{result.Error.Message.GetValueOrEmptyString()}", result.Error.FailureType);
+                            ExternalReference externalReference;
+                            if (externalReferenceProperties.Uuid.HasValue)
+                            {
+                                var uuid = externalReferenceProperties.Uuid.Value;
+                                var existingReferenceResult = _referenceRepository.GetByUuid(uuid);
+                                if(existingReferenceResult.IsNone)
+                                    return new OperationError($"External reference with uuid: {uuid} was not found", OperationFailure.NotFound);
```

Comment:
would be nice with a reference to the rootId in the error message :-)
Also change the code to BadInput.. The operation works on "the root" and in that context this error is badinput... not finding the root would be notfound :-)